### PR TITLE
FIX:CAP-45 - Fields should be required on Signup and Login Pages

### PR DESCRIPTION
**Pull request by rooshnie23**
## Description

**JIRA Ticket:** [CAP-45](https://mfslcapstone.atlassian.net/browse/CAP-45)

**Summary:**
The fields must have elements for the sign up and login buttons to be enabled. There's also an addition feature where the user must enter a 8 characters long password. If the latter is not respected, then the button will be disabled. Also, fixed the issue with the text field titles getting cutoff.

**Screenshots/Examples:**
image
image
image
image
image
**For Login:**
image
image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-functional change which does not add or remove functionality)

## Checklist

- [X] My branch and PR name follow the [naming conventions](https://mfslcapstone.atlassian.net/wiki/spaces/SD/pages/4358145/Development+Naming+Conventions) the team has established.
- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [X] I have added tests that prove my fix is effective or that my feature works.
- [X] New and existing unit tests pass locally with my changes.
- [X] Any dependent changes have been merged and published in downstream modules. (If not, this should be a draft PR)
- [X] I have [rebased](https://mfslcapstone.atlassian.net/wiki/spaces/SD/pages/4456479/How+To%3A+Rebase+Branch+Onto+Main) my branch onto main before opening this PR, and resolved any merge conflicts.
- [X] I have ensured that new and existing integrations work with my changes.

[CAP-45]: https://mfslcapstone.atlassian.net/browse/CAP-45?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Pages/SignUpPage/SignUpPage.tsx [hunk 1]
```diff
@@ -22,7 +35,7 @@ const SignUpPage: React.FC = () => {
                 <TextField
                     required
                     label="Name"
```
Comment (abhijeetpraveen) [inline src/Pages/SignUpPage/SignUpPage.tsx:37]: @rooshnie23 Don't think we need a name field in the Signup page. In the backend we only have the username and the password when signing up. I don't think we need to include it. It is not being used anywhere in the application too.

File: src/Pages/SignUpPage/SignUpPage.tsx [hunk 2]
```diff
@@ -8,9 +8,22 @@ const SignUpPage: React.FC = () => {
     const [username, setUsername] = useState('');
     const [password, setPassword] = useState('');
     const [isCheckboxChecked, setIsCheckboxChecked] = useState(false);
-
+    const [passwordError, setPasswordError] = useState('');
+    const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const { value } = e.target;
+        setPassword(value);
```
Comment (abhijeetpraveen) [inline src/Pages/SignUpPage/SignUpPage.tsx:14]: @rooshnie23 The password should be set after the check for the value being 8 characters is complete

File: src/Pages/SignUpPage/SignUpPage.tsx [hunk 3]
```diff
@@ -41,28 +54,34 @@ const SignUpPage: React.FC = () => {
...
                                   
-                <Button variant="contained"
-                        className="sign-up-button"
-                        onClick={handleLogin}
-                        disabled={!isCheckboxChecked}
+                <Button
+                    variant="contained"
+                    className="sign-up-button"
+                    onClick={handleLogin}
+                    disabled={!isCheckboxChecked || !isFormFilled()}
```
Comment (abhijeetpraveen) [inline src/Pages/SignUpPage/SignUpPage.tsx:84]: @rooshnie23 The button should also be disabled if the password error is not an empty string

**Review by abhijeetpraveen — CHANGES_REQUESTED** [review]

File: src/Pages/SignUpPage/SignUpPage.tsx [hunk 4]
```diff
@@ -8,9 +8,22 @@ const SignUpPage: React.FC = () => {
...
     const [isCheckboxChecked, setIsCheckboxChecked] = useState(false);
-
+    const [passwordError, setPasswordError] = useState('');
+    const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const { value } = e.target;
+        setPassword(value);
+        if (value.length < 8) {
+            setPasswordError('Password must be at least 8 characters long');
+        } else {
+            setPasswordError('');
```
Comment (abhijeetpraveen) [inline src/Pages/SignUpPage/SignUpPage.tsx:18]: @rooshnie23 you could set the password within this else statement or outside this if-else block altogether

**Review by abhijeetpraveen — CHANGES_REQUESTED** [review]

File: src/Pages/SignUpPage/SignUpPage.tsx (hunk 2 above)
Comment (rooshnie23) [reply src/Pages/SignUpPage/SignUpPage.tsx:14]: It's reactive, so the password is constantly set. In other words, the final password is set when the user stops writing their password. And it only handles the login, after the button is pressed and both text fields are filled with their appropriate requirements.

File: src/Pages/SignUpPage/SignUpPage.tsx (hunk 3 above)
Comment (rooshnie23) [reply src/Pages/SignUpPage/SignUpPage.tsx:84]: Could you elaborate on this? I don't seem to understand.

File: src/Pages/SignUpPage/SignUpPage.tsx (hunk 2 above)
Comment (rooshnie23) [reply src/Pages/SignUpPage/SignUpPage.tsx:14]: Another thing, the same happens for when the password is below 8 characters, it is constantly getting checked. If you tried it, the password error shows as soon as the user writes the first character.

File: src/Pages/SignUpPage/SignUpPage.tsx (hunk 1 above)
Comment (rooshnie23) [reply src/Pages/SignUpPage/SignUpPage.tsx:37]: Definitely, I'll bring the change to it.

File: src/Pages/SignUpPage/SignUpPage.tsx (hunk 3 above)
Comment (abhijeetpraveen) [reply src/Pages/SignUpPage/SignUpPage.tsx:84]: If we see that the passwordError is not empty, which means the password isn't 8 characters yet, then the button should be disabled as well. It should only get enabled once everything is filled out as well as the password having 8 characters.

File: src/Pages/SignUpPage/SignUpPage.tsx (hunk 2 above)
Comment (abhijeetpraveen) [reply src/Pages/SignUpPage/SignUpPage.tsx:14]: I see, that's perfect. Thanks for letting me know. Didn't know that

File: src/Pages/SignUpPage/SignUpPage.tsx (hunk 1 above)
Comment (rooshnie23) [reply src/Pages/SignUpPage/SignUpPage.tsx]: Done!

**Review by abhijeetpraveen — APPROVED** [review]
Looks good to me! You can merge it now.

**Pull request merged**